File: python/code-duplication/good/order_processor.py
class OrderProcessor:
    def __init__(self):
        self.tax_calculator = TaxCalculator()
        self.shipping_calculator = ShippingCalculator()

    def calculate_tax(self, price: float, state: str) -> float:
        return self.tax_calculator.calculate_tax(price, state)

    def calculate_shipping(self, weight: float, distance: float) -> float:
        return self.shipping_calculator.calculate_shipping(weight, distance)


class InvoiceGenerator:
    def __init__(self):
        self.tax_calculator = TaxCalculator()
        self.shipping_calculator = ShippingCalculator()

    def calculate_tax(self, price: float, state: str) -> float:
        return self.tax_calculator.calculate_tax(price, state)

    def calculate_shipping(self, weight: float, distance: float) -> float:
        return self.shipping_calculator.calculate_shipping(weight, distance)


class QuoteGenerator:
    def __init__(self):
        self.tax_calculator = TaxCalculator()
        self.shipping_calculator = ShippingCalculator()

    def calculate_tax(self, price: float, state: str) -> float:
        return self.tax_calculator.calculate_tax(price, state)

    def calculate_shipping(self, weight: float, distance: float) -> float:
        return self.shipping_calculator.calculate_shipping(weight, distance)

File: code-duplication/good/OrderProcessor.php
<?php

class OrderProcessor {
    private TaxCalculator $taxCalculator;
    private ShippingCalculator $shippingCalculator;

    public function __construct() {
        $this->taxCalculator = new TaxCalculator();
        $this->shippingCalculator = new ShippingCalculator();
    }

    public function calculateTax(float $price, string $state): float {
        return $this->taxCalculator->calculateTax($price, $state);
    }

    public function calculateShipping(float $weight, float $distance): float {
        return $this->shippingCalculator->calculateShipping($weight, $distance);
    }
}

class InvoiceGenerator {
    private TaxCalculator $taxCalculator;
    private ShippingCalculator $shippingCalculator;

    public function __construct() {
        $this->taxCalculator = new TaxCalculator();
        $this->shippingCalculator = new ShippingCalculator();
    }

    public function calculateTax(float $price, string $state): float {
        return $this->taxCalculator->calculateTax($price, $state);
    }

    public function calculateShipping(float $weight, float $distance): float {
        return $this->shippingCalculator->calculateShipping($weight, $distance);
    }
}

class QuoteGenerator {
    private TaxCalculator $taxCalculator;
    private ShippingCalculator $shippingCalculator;

    public function __construct() {
        $this->taxCalculator = new TaxCalculator();
        $this->shippingCalculator = new ShippingCalculator();
    }

    public function calculateTax(float $price, string $state): float {
        return $this->taxCalculator->calculateTax($price, $state);
    }

    public function calculateShipping(float $weight, float $distance): float {
        return $this->shippingCalculator->calculateShipping($weight, $distance);
    }
}
